### DISCUSSION TITLE
Error compiling components when Vue 2 is installed in main package.json

### DISCUSSION THREAD

**Discussion by WillyReyno** [laravel/nova-issues#4678] [Q&A] [answered]
- Laravel Version: 9.24.0
- Nova Version: 4.12.14
- PHP Version: 8.0.9
- Operating System and Version: MacOS 12.5
- Reproduction Repository: https://github.com/WillyReyno/nova-component-vue

### Description:

This is issue is related to discussion #4549, but I'm creating an issue as I was able to make a reproduction repo.

When Vue 2 is installed in the main package.json of the laravel app, nova components compilation fails with the following error: 

![Capture d’écran 2022-08-09 à 19 24 24](https://user-images.githubusercontent.com/5576482/183717948-3e6ed012-2dd7-459d-9351-97b53eed70a0.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:

* Create custom tool `php artisan nova:tool acme/price-tracker`
* Install Vue 2 in project's root : `npm install vue@^2.7.8`
* Move into custom tool directory `cd nova-components/PriceTracker`
* Run `npm run dev` / `npm run prod`

**Comment by crynobone** [comment]
This is either an issue with Webpack, Mix or NPM. So far we haven't found a solution for it.

**Comment by crynobone** [comment, marked as answer]
After some additional research, this is due to `vue-loader` trying to use `require('vue/compiler-sfc')` insteads of `require('@vue/compiler-sfc')` which is required directly in the template. https://github.com/vuejs/vue-loader/blob/6a293dff5f2ed3167eb3b448f1d714091185a44e/src/compiler.ts#L12-L25

This can be solved by adding the following:

```bash
npm uninstall "@vue/compiler-sfc"
npm install -D "vue"
```

Example: https://github.com/nova-issues/issue-4595/commit/e56d8af20ed37ef4a1ce4ac0a8596852d8f15c07

There could be side-effect on requiring `vue` directly (the generated dist are slightly larger with more code) but I can't find any error at the moment. Best to first try with `npm run dev` and tests it out before using `npm run prod`.
